packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js

```diff
@@ -0,0 +1,139 @@
+import generatedAnswerTemplate from './generatedAnswer.html';
+// @ts-ignore
+import loadingTemplate from './loading.html';
+
+/** @typedef {import("coveo").SearchEngine} SearchEngine */
+/** @typedef {import("coveo").GeneratedAnswer} GeneratedAnswer */
+/** @typedef {import("coveo").GeneratedAnswerState} GeneratedAnswerState */
+/** @typedef {import("coveo").GeneratedAnswerCitation} GeneratedAnswerCitation */
+
+/**
+ * The `QuanticGeneratedAnswer` component automatically generates an answer using Coveo machine learning models to answer the query executed by the user.
+ * @category Search
+ * @example
+ * <c-quantic-generated-answer engine-id={engineId}></c-quantic-generated-answer>
+ */
+export default class QuanticGeneratedAnswer extends LightningElement {
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+
+  labels = {
+    generatedAnswerForYou,
+    loading,
+  };
+
+  /** @type {GeneratedAnswer} */
+  generatedAnswer;
+  /** @type {GeneratedAnswerState} */
+  state;
+  /** @type {'neutral' | 'liked' | 'disliked'} */
+  feedbackState = 'neutral';
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  /**
+   * @param {SearchEngine} engine
+   */
+  initialize = (engine) => {
+    this.headless = getHeadlessBundle(this.engineId);
+    this.generatedAnswer = this.headless.buildGeneratedAnswer(engine);
+    this.unsubscribeGeneratedAnswer = this.generatedAnswer.subscribe(() =>
+      this.updateState()
+    );
+  };
+
+  disconnectedCallback() {
+    this.unsubscribeGeneratedAnswer?.();
+  }
+
+  updateState() {
+    this.state = this.generatedAnswer.state;
+    this.updateFeedbackState();
+  }
+
+  updateFeedbackState() {
+    if (this.state?.liked) {
+      this.feedbackState = 'liked';
+    } else if (this.state?.disliked) {
+      this.feedbackState = 'disliked';
+    } else {
+      this.feedbackState = 'neutral';
+    }
+  }
+
+  /**
+   * handles clicking on a citation.
+   * @param {string} id
+   */
+  handleCitationClick = (id) => {
+    this.generatedAnswer.logCitationClick(id);
+  };
+
+  /**
+   * handles liking the generated answer.
+   * @param {CustomEvent} event
+   */
+  handleLike(event) {
+    event.stopPropagation();
+    this.generatedAnswer.like?.();
+  }
+
+  /**
+   * handles disliking the generated answer.
+   * @param {CustomEvent} event
+   */
+  handleDislike(event) {
+    event.stopPropagation();
+    this.generatedAnswer.dislike?.();
+  }
+
+  get answer() {
+    return this?.state?.answer;
+  }
+
+  get citations() {
+    return this?.state?.citations;
+  }
+
+  get isLoading() {
+    return this?.state?.isLoading;
+  }
+
+  get isStreaming() {
+    return this?.state?.isStreaming;
+  }
+
+  get shouldDisplayGeneratedAnswer() {
+    return !!this.answer || this.isStreaming || !!this.citations?.length;
```

Comment:
Just a question, if ML returns for some reasons citations but no answer it will still display the feedback and citations? Wondering if we want citations.length here for that reason, unless its "near zero" chances of happening ;D

Reply:
Yea it's indeed weird to see that part of the condition? Is it possible that `this.answer = false; this.isStreaming = false` but `this.citations.length > 0`?

Also if you are using `.length`, please use `this.citations?.length > 0`.

Reply:
It's indeed something that's not supposed to happen and probably will never happen. `!!this.answer` and `this.citations?.length > 0` are two equivalent boolean expressions and will always have the same boolean value.
For this reason we can completely get rid of `!!this.citations?.length` , but I'm keeping it just because it makes the component easier to test since I will just need to mock the response that returns the citations in order to test that they are displayed properly.